solr/core/src/java/org/apache/solr/servlet/RateLimitManager.java

```diff
@@ -105,6 +130,10 @@ public RequestRateLimiter.SlotReservation handleRequest(HttpServletRequest reque
       return RequestRateLimiter.UNLIMITED;
     }
 
+    if (typeOfRequest.equals(SolrRequest.RequestPriorities.FOREGROUND.name())
+        || typeOfRequest.equals(SolrRequest.RequestPriorities.BACKGROUND.name())) {
+      typeOfRequest = SolrRequest.SolrRequestType.PRIORITY_BASED.name();
+    }
```

Comment:
This is weird IMO. I think clients wanting to use PriorityBased ratelimiter should specify request type just as they would for any other ratelimiter (i.e. in this case `PRIORITY_BASED`), and should pass any extra information (e.g. `FOREGROUND` or `BACKGROUND` in custom headers dedicated to that purpose (e.g., `Solr-Request-Priority: FOREGROUND`). The approach I'm suggesting here would be clearer, and would allow cleaner separation of concerns.